korge/src/commonMain/kotlin/com/soywiz/korge/view/Circle.kt

```diff
@@ -8,37 +9,39 @@ import com.soywiz.korma.geom.vector.*
  * The [autoScaling] determines if the underlying texture will be updated when the hierarchy is scaled.
  * The [callback] allows to configure the [Circle] instance.
  */
-inline fun Container.circle(radius: Double = 16.0, color: RGBA = Colors.WHITE, autoScaling: Boolean = true, callback: Circle.() -> Unit = {}): Circle = Circle(radius, color, autoScaling).addTo(this).apply(callback)
+inline fun Container.circle(
+    radius: Double = 16.0,
+    color: RGBA = Colors.WHITE,
+    autoScaling: Boolean = true,
+    callback: Circle.() -> Unit = {}
+): Circle = Circle(radius, color, autoScaling).addTo(this).apply(callback)
 
 /**
  * A [Graphics] class that automatically keeps a circle shape with [radius] and [color].
  * The [autoScaling] property determines if the underlying texture will be updated when the hierarchy is scaled.
  */
-open class Circle(radius: Double = 16.0, color: RGBA = Colors.WHITE, autoScaling: Boolean = true) : Graphics(autoScaling = autoScaling) {
+open class Circle(
+    radius: Double = 16.0,
+    color: RGBA = Colors.WHITE,
+    autoScaling: Boolean = true
+) : Graphics(autoScaling = autoScaling) {
+
     /** Radius of the circle */
-    var radius: Double = radius
-        set(value) {
-            field = value; updateGraphics()
-        }
+    var radius: Double by uiObservable(radius) { updateGraphics() }
+    /** Color of the circle */
+    var color: RGBA by uiObservable(color) { updateGraphics() }
 
     override val bwidth get() = radius * 2
     override val bheight get() = radius * 2
 
-    /** Color of the circle */
-    // @TODO: We could paint a white circle, and then tint it using the [colorMul] property
-    var color: RGBA = color
-        set(value) {
-            field = value; updateGraphics()
-        }
-
     init {
         updateGraphics()
     }
 
     private fun updateGraphics() {
         clear()
         fill(color) {
-            circle(0.0, 0.0, radius)
+            circle(radius, radius, radius)
```

Comment:
I don't think that's the right position. Unless the circle/ellipse was provided by a bounds, I think that by the principle of least surprise it should be at `[0, 0]`. But indeeed if this failed, somewhere else there might be a bug

Reply:
Some time ago I tried to use `Circle` view, but found this error - wrong position of `Circle`. I needed to call `position(radius, radius)` by myself in order to make it fully visible. So I think its position should be like now.

Reply:
When drawing `circle` inside your own `Graphics` view, positioning circle in [0; 0] is correct. But as soon as you use `Circle` view, this view's top left corner should be positioned in [0; 0], not its center.

Reply:
I see your point. Since Graphics extends `RectBase` it seems it have `anchorX` and `anchorY` and extension methods:

```
inline fun <T : RectBase> T.anchor(ax: Number, ay: Number): T =
	this.apply { this.anchorX = ax.toDouble() }.apply { this.anchorY = ay.toDouble() }

fun <T : RectBase> T.center(): T = anchor(0.5, 0.5)
val <T : RectBase> T.centered: T get() = anchor(0.5, 0.5)
```

So I guess that's fine. People can use `.centered` if required.